### PR TITLE
feat: support typescript 4.9, drop support for typescript 4.2, 4.3

### PR DESCRIPTION
**Pull request by wtchnm**
#1462

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -146,13 +146,13 @@
...
     "tsup": "^5.12.8",
-    "typescript": "^4.8.2",
+    "typescript": "^4.9.3",
     "url-loader": "^4.1.1",
     "webpack": "^5.68.0",
     "webpack-dev-server": "^3.11.2"
   },
   "peerDependencies": {
-    "typescript": ">= 4.2.x <= 4.8.x"
+    "typescript": ">= 4.4.x <= 4.9.x"
```
Comment (kettanaito) [inline package.json:155]: Do you happen to know how old TS 4.2/4.3 is? I wonder if it's safe to drop support for them.

Comment (wtchnm) [reply package.json:155]: TS 4.2 is 1 year and 9 months old, TS 4.3 is 1 year and 6 months old. According to the release notes, no breaking changes would block migration to TS 4.4.
See [TS 4.3 breaking changes](https://devblogs.microsoft.com/typescript/announcing-typescript-4-3/#breaking-changes) and [TS 4.4 breaking changes](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#breaking-changes).

Comment (kettanaito) [reply package.json:155]: Thanks for looking that up! That's lovely, let's drop their support and introduce 4.9.

**Review by kettanaito — APPROVED** [review]
This is great. Thanks for working on this, @wtchnm 👏

**Comment by wtchnm** [comment]
Sure, thanks for your amazing work on this lib!

**Comment by kettanaito** [comment]
## Released: v0.49.0 🎉

This has been released in v0.49.0!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v0.49.0)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/0.49.0)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Comment by andykenward** [comment]
@kettanaito Can you also do this for the msw@next `0.0.0-fetch.rc-1` ? I'm currently getting this npm error.

node : `16.18.1`
npm: `8.19.2`

```
$ npm install msw@next
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: msw@0.0.0-fetch.rc-1
npm ERR! Found: typescript@4.9.3
npm ERR! node_modules/typescript
npm ERR!   typescript@"4.9.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peerOptional typescript@">= 4.2.x <= 4.8.x" from msw@0.0.0-fetch.rc-1
npm ERR! node_modules/msw
npm ERR!   msw@"0.0.0-fetch.rc-1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: typescript@4.8.4
npm ERR! node_modules/typescript
npm ERR!   peerOptional typescript@">= 4.2.x <= 4.8.x" from msw@0.0.0-fetch.rc-1
npm ERR!   node_modules/msw
npm ERR!     msw@"0.0.0-fetch.rc-1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

**Comment by kettanaito** [comment]
Hey, @andykenward. Sure, will update that release as well when I have a minute.

**Pull request merged**